### PR TITLE
fix: incorrect node resources

### PR DESCRIPTION
**Pull request by wangyizhi1**
#### What type of PR is this?

/kind bug

#### What does this PR do?

fix: incorrect node resources

#### Which issue(s) does this PR fix?

Fixes #
#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Comment by wangyizhi1** [comment]
@duanmengkk @hanweisen

File: pkg/utils/resources.go [hunk 1]
```diff
@@ -70,3 +77,18 @@ func GetPodsTotalRequestsAndLimits(podList *corev1.PodList) (reqs corev1.Resourc
 	}
 	return
 }
+
+func GetUsedPodNums(podList *corev1.PodList) (res corev1.ResourceList) {
+	podQuantity := resource.Quantity{}
+	res = corev1.ResourceList{}
+	for _, p := range podList.Items {
+		pod := p
```
Comment (hanweisen) [inline pkg/utils/resources.go:85]: In the loop, A new variable `pod` to store the current Pod, which is unnecessary. should directly use `p`.

File: pkg/utils/resources.go [hunk 2]
```diff
@@ -70,3 +77,18 @@ func GetPodsTotalRequestsAndLimits(podList *corev1.PodList) (reqs corev1.Resourc
 	}
 	return
 }
+
+func GetUsedPodNums(podList *corev1.PodList) (res corev1.ResourceList) {
```
Comment (hanweisen) [inline pkg/utils/resources.go:81]: May this is better avoid pod object copy
```
func GetUsedPodNums(podList *corev1.PodList) (res corev1.ResourceList) {
	podQuantity := resource.Quantity{}
	res = corev1.ResourceList{}
	for i := range podList.Items {
		if IsVirtualPod(&podList.Items[i) {
			continue
		}
		q := resource.MustParse("1")
		podQuantity.Add(q)
	}
	res[podResourceName] = podQuantity
	return
}
```

Comment (hanweisen) [reply pkg/utils/resources.go:81]: Additionally, I think we can put `q := resource.MustParse("1")` out of the loop

Comment (wangyizhi1) [reply pkg/utils/resources.go:81]: Each real pod needs to execute ``resource.MustParse("1")`` once

**Pull request merged**